### DISCUSSION TITLE
Beginner Questions

### DISCUSSION THREAD

**Discussion by cjordog** [VerySleepy/verysleepy#106] [Q&A]
Hi. I have a couple of beginner questions. Does VerySleepy monitor spawned threads from a launched application? Or does it only profile the main thread of that launched application. If the latter, is there a way to get it to pick up spawned threads?

I'm also seeing some missing symbols on some of my functions. Most of them are there, but I'm trying to track down where these last few missing ones might be coming from. A few of them are missing symbols from KERNEL modules. Do i need any specific windows tools or anything to see these? Otherwise, I'm lost as to why only a few of them are missing.
![image](https://user-images.githubusercontent.com/15066247/190974736-c5e0fc12-ed39-4b0a-9b34-0723751e1e46.png)

**Comment by CyberShadow** [comment]
> Hi. I have a couple of beginner questions. Does VerySleepy monitor spawned threads from a launched application? Or does it only profile the main thread of that launched application. If the latter, is there a way to get it to pick up spawned threads?

Using the [latest version from Git](https://github.com/VerySleepy/verysleepy/releases/tag/appveyor), the following should be accurate:

> If no threads are selected, Very Sleepy will profile all threads, and attach to the process as a debugger to profile new threads as they are created. To prevent this (e.g. if you wish to attach another debugger during profiling), simply select all threads.

---

> I'm also seeing some missing symbols on some of my functions. Most of them are there, but I'm trying to track down where these last few missing ones might be coming from. A few of them are missing symbols from KERNEL modules. Do i need any specific windows tools or anything to see these? Otherwise, I'm lost as to why only a few of them are missing.

I don't know, do you have the Microsoft symbol server configured?

**Reply by cjordog** [reply]
I can see all application threads if I inspect the application from the main menu. But if I launch an application from VerySleepy, it doesnt seem to pick up new threads. Maybe I'm missing an option somewhere?
And yes, I believe the default settings have the microsoft symbol server configured, unless theres extra steps i need to take.

**Reply by CyberShadow** [reply]
> Maybe I'm missing an option somewhere?

It's possible.

> And yes, I believe the default settings have the microsoft symbol server configured, unless theres extra steps i need to take.

It's possible that `dbghelpms.dll` needs to be updated again with the latest `dbghelp.dll`. Maybe you could check what versions are available on your system or can be installed, and see if the latest version you can find fares any better.

I think the Log panel should also print some messages about symbol acquisition.
